Objects/moduleobject.c

```diff
@@ -699,6 +699,33 @@ module_repr(PyModuleObject *m)
     return PyObject_CallMethod(interp->importlib, "_module_repr", "O", m);
 }
 
+int
+_PyModule_IsInitializing(PyObject *m)
+{
+    if (PyModule_Check(m) && ((PyModuleObject *)m)->md_dict) {
+        _Py_IDENTIFIER(__spec__);
+        _Py_IDENTIFIER(_initializing);
+        PyObject *value = NULL;
+        PyObject *spec;
+        int initializing = 0;
+        spec = _PyDict_GetItemId(((PyModuleObject *)m)->md_dict, &PyId___spec__);
+        if (spec != NULL && spec != Py_None) {
+            value = _PyObject_GetAttrId(spec, &PyId__initializing);
+        }
+        if (value == NULL) {
+            PyErr_Clear();
+        }
+        else {
+            initializing = PyObject_IsTrue(value);
+            Py_DECREF(value);
+            if (initializing < 0)
+                PyErr_Clear();
```

Comment:
Curly braces around the statement.

Reply:
What if test `value == Py_True` instead of `PyObject_IsTrue(value)`? This would simplify the code, and I don't think that we should expect any true value except `True`.

Reply:
I'm torn. I do agree the code is simpler and you shouldn't expect it, but I also don't know if it's worth restricting the expectations that tightly. Then again, who is going to set this other than import itself?

IOW I don't have an opinion so whatever you prefer. 😄

Reply:
And please note, that this PR changes the behavior of `PyImport_ImportModuleLevelObject()`. I have extracted the common code into `_PyModule_IsInitializing()`, but the code in `PyImport_ImportModuleLevelObject()` is not exactly the same as in `module_getattro()`. It used `_PyObject_GetAttrId(mod, &PyId___spec__)`, but now it uses `_PyDict_GetItemId(((PyModuleObject *)m)->md_dict, &PyId___spec__)`, and supports only the module type instances. I.e. `__getattr__()` will be now bypassed when looking up `__spec__`.

Reply:
That is a breaking changing due to the fact that people could return any object for their module thanks to `importlib.abc.Loader.create_module()`.